useful-scripts/permalink-snapshot-migration/migrate_legacy_permalinks.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import csv
+import os
+import time
+
+import requests
+
+"""
+This is the script for migrating legacy permalinks to permalink snapshots.
+It uses a Data API migration endpoint `PUT /api/permalink/{permalinkId}/snapshot`.
+It reads a CSV file named 'permalinks.csv' and outputs a CSV file named 'migration_results.csv'.
+
+Usage: `pipenv run python migrate_legacy_permalinks.py`
+
+Instructions:
+
+1. List all the legacy permalinks by running a query against the Content database to get all the legacy permalinks:
+
+```
+SELECT LOWER(Id) AS permalink_id,
+       IIF(LegacyHasSnapshot = 1, 'True', 'False') AS migrated
```

Comment:
Just wondering if this is a typo? `IIF`

Reply:
It's not a typo, it's shorthand for a case statement, see [IIF](https://learn.microsoft.com/en-us/sql/t-sql/functions/logical-functions-iif-transact-sql?view=sql-server-ver16).